packages/patternfly-4/react-core/src/components/Switch/Switch.tsx

```diff
@@ -47,14 +46,14 @@ export class Switch extends React.Component<SwitchProps> {
   }
 
   render() {
-    const { className, label, isChecked, isDisabled, onChange, ...props } = this.props;
+    const { className, label, isChecked, isDisabled, onChange, ouiaContext, ...props } = this.props;
     return (
       <label
         className={css(styles.switch, className)}
         htmlFor={this.id}
-        {...isOUIAEnvironment() && {
+        {...ouiaContext.isOuia && {
           'data-ouia-component-type': 'Switch',
-          'data-ouia-component-id': this.ouiaId
+          'data-ouia-component-id': ouiaContext.ouiaId
```

Comment:
Can we remove `ouiaId` from context and consume it from props instead? It would be kinda repetetive to add id to multiple items trough context provider, plus not all items will have `data-ouia-component-id` set. Only if there will be multiple items of same component we will use `ouiaId`

Reply:
ok, i'll make it so it can be consumed from props